pennylane/gradients/general_shift_rules.py

```diff
@@ -46,38 +54,26 @@ def process_shifts(rule, tol=1e-10, check_duplicates=True):
     - Finally, the terms are sorted according to the absolute value of ``shift``,
       This ensures that a zero-shift term, if it exists, is returned first.
     """
-    # remove all small coefficients and shifts
+    # set all small coefficients, multipliers if present, and shifts to zero.
     rule[np.abs(rule) < tol] = 0
 
     # remove columns where the coefficients are 0
-    rule = rule[:, ~(rule[0] == 0)]
+    rule = rule[~(rule[:, 0] == 0)]
 
-    if check_duplicates:
+    if batch_duplicates:
         round_decimals = int(-np.log10(tol))
-        rounded_rule = np.round(rule[1:], round_decimals)
+        rounded_rule = np.round(rule[:, 1:], round_decimals)
         # determine unique shifts or (multiplier, shift) combinations
-        unique_mods = np.unique(rounded_rule, axis=1)
-
-        if rule.shape[-1] != unique_mods.shape[-1]:
-            if rule.shape[0] == 3:
-                # sum columns that have the same multiplier and shift value
-                # TODO: Check whether the list comprehension can be absorbed into a smart
-                # single NumPy call
-                coeffs = [
-                    np.sum(rule[0, np.all(rounded_rule[1:] == mod.reshape((2, 1)), axis=0)])
-                    for mod in unique_mods.T
-                ]
-            else:
-                # sum columns that have the same shift value
-                rounded_rule = rounded_rule[0]
-                # TODO: Check whether the list comprehension can be absorbed into a smart
-                # single NumPy call
-                coeffs = [np.sum(rule[0, rounded_rule == s]) for s in unique_mods[0]]
-
-            rule = np.vstack([np.stack(coeffs), unique_mods])
+        unique_mods = np.unique(rounded_rule, axis=0)
+
+        if rule.shape[0] != unique_mods.shape[0]:
+            matches = np.all(rounded_rule[:, np.newaxis] == unique_mods[np.newaxis, :], axis=-1)
+            # TODO: The following line probably can be done in numpy
+            coeffs = [np.sum(rule[slc, 0]) for slc in matches.T]
+            rule = np.hstack([np.stack(coeffs)[:, np.newaxis], unique_mods])

```

Comment:
This simplification is simply due to being able to transpose?!

Reply:
No, this simplification is in addition. I meddled a bit with numpy and found this pipeline that does not care whether there are multipliers or not :)

Reply:
oh! nice find ;)